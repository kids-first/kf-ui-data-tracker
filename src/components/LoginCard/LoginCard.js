import React from 'react';
import GoogleLogin from 'react-google-login';
import { GOOGLE_APP_ID, EGO_API } from '../../common/globals';
import jwtDecode from 'jwt-decode';
import { ApolloConsumer } from 'react-apollo';
import { Button, Icon } from 'semantic-ui-react';
import { withRouter } from 'react-router';
import { auth } from '../../state/auth';
import Auth0_logo from '../../assets/Auth0_logo.png';

const onSuccess = (repsonse, client, history) => {
  fetch(EGO_API + '/oauth/google/token', {
    method: 'GET',
    mode: 'cors',
    headers: {
      token: repsonse.tokenId,
      'content-type': 'application/json',
    },
  })
    .then(resp => {
      return resp.text();
    })
    .then(text => {
      localStorage.setItem('egoToken', text);

      const jwtData = jwtDecode(text);
      const user = { ...jwtData.context };
      client.writeData({ data: { user } });

      history.push('/');

      return text;
    })
    .catch(err => {
      console.log('Problem getting Ego token');
      console.log(err);
    });
};

const onFailure = repsonse => {
  console.log('Problem sign in');
};

const LoginContainer = ({ history }) => (
  <ApolloConsumer>
    {client => (
      <Button.Group fluid textAlign="center">
        <Button
          className="button--login"
          onClick={() => auth.login()}
          size="large"
          color="black"
        >
          <img className="auth0-logo" src={Auth0_logo} alt="Auth0 logo" />
        </Button>
        <Button.Or className="button--login" />
        <GoogleLogin
          clientId={GOOGLE_APP_ID}
          buttonText="Sign in with Google"
          onSuccess={response => onSuccess(response, client, history)}
          onFailure={onFailure}
          render={renderProps => (
            <Button
              className="button--login"
              onClick={() => renderProps.onClick}
              size="large"
              primary
            >
              <Icon name="google" />
              Google
            </Button>
          )}
        />
      </ Button.Group >
    )}
  </ApolloConsumer>
);

export default withRouter(LoginContainer);
