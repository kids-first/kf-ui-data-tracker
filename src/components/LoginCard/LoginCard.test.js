import React from 'react';
import wait from 'waait';
import {MockedProvider} from 'react-apollo/test-utils';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'react-testing-library';
import LoginCard from './LoginCard';

jest.mock('auth0-js');

it('renders correctly', async () => {
  jest.mock('../../state/auth');
  const tree = render(
    <MockedProvider>
      <MemoryRouter initialEntries={['/login']}>
        <LoginCard />
      </MemoryRouter>
    </MockedProvider>,
  );
  await wait(0);
  expect(tree).toMatchSnapshot();
});
