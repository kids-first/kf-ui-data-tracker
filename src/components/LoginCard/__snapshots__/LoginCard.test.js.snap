// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui large vertical buttons"
      >
        <button
          class="ui icon positive right labeled button"
        >
          Login with Auth0
          <i
            aria-hidden="true"
            class="chevron right icon"
          />
        </button>
        <button
          class="ui icon right labeled button"
        >
          Login with Ego
          <i
            aria-hidden="true"
            class="chevron right icon"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui large vertical buttons"
    >
      <button
        class="ui icon positive right labeled button"
      >
        Login with Auth0
        <i
          aria-hidden="true"
          class="chevron right icon"
        />
      </button>
      <button
        class="ui icon right labeled button"
      >
        Login with Ego
        <i
          aria-hidden="true"
          class="chevron right icon"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
